qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -471,9 +470,9 @@ def _assert_mode(self, mode: str) -> None:
         Args:
             mode: "CURR" or "VOLT"
         """
-        if self._cached_mode != mode:
+        if self.source_mode() != mode:
```

Comment:
why does this not use get_latest ? 

The docstring is also wrong (There is no such thing as check)

Reply:
get_latest is added.
The docstring could be something like this: To raise Error when the active source_mode of the instrument is not the same as the desired mode of the user.

Reply:
I would just delete the second line of the docstring. The first line is fine